### ISSUE TITLE
Z3.SMT.parse_smtlib2_string seg fault

### ISSUE THREAD

**Issue by anmaped** [Z3Prover/z3#1831]
With the current git version I got 
```
Problem:

Segmentation fault
```
when executing
```
let ctx = (mk_context [("model", "true")]) in
let exp = parse_smtlib2_string ctx smtlibv2_str [] [] [] [] in
print_endline "\nProblem:\n" ; print_endline (Z3.AST.to_string (ast_of_expr exp)) ;
```
no matter what is the smtlibv2_str. This can be reproducible in linux and windows (mingw32).

It works well with z3 version 7c63a5cc1ddf20e769a6d9abb1a43f561ca1ddf5 and 3b1b82bef05a1b5fd69ece79c80a95fb6d72a990.

**Comment by NikolajBjorner** [comment]
The signature wasn't updated when the C-based API was changed. It should be AST.ASTVector.ast_vector.

**Comment by NikolajBjorner** [comment]
Fix presumably with a5762a78e94dea0d02ba101692b68b847fa72195

**Comment by NikolajBjorner** [comment]
closing as this would be fixed by the above update.

**Comment by anmaped** [comment]
The ML API is not building with the current git version.

```
ocamlfind ocamlc -package num  -i -I api/ml -c ../src/api/ml/z3enums.ml > api/ml/z3enums.mli
ocamlfind ocamlc -package num  -I api/ml -o api/ml/z3enums.cmi -c api/ml/z3enums.mli
ocamlfind ocamlc -package num  -I api/ml -o api/ml/z3enums.cmo -c ../src/api/ml/z3enums.ml
ocamlfind ocamlc -package num  -i -I api/ml -c ../src/api/ml/z3native.ml > api/ml/z3native.mli
ocamlfind ocamlc -package num  -I api/ml -o api/ml/z3native.cmi -c api/ml/z3native.mli
ocamlfind ocamlc -package num  -I api/ml -o api/ml/z3native.cmo -c ../src/api/ml/z3native.ml
cp ../src/api/ml/z3.mli api/ml/z3.mli
ocamlfind ocamlc -package num  -I api/ml -o api/ml/z3.cmi -c api/ml/z3.mli
ocamlfind ocamlc -package num  -I api/ml -o api/ml/z3.cmo -c ../src/api/ml/z3.ml
File "../src/api/ml/z3.ml", line 1950, characters 12-24:
Error: This function has type int -> Z3enums.lbool
       It is applied to too many arguments; maybe you forgot a `;'.
make: *** [Makefile:4370: api/ml/z3.cmo] Error 2
```

**Comment by anmaped** [comment]
@NikolajBjorner Thanks for the fix a5762a7. It is working now.
